src/main/groovy/edu/oregonstate/mist/invencycle/resources/BikeResource.groovy

```diff
@@ -93,6 +94,40 @@ class BikeResource extends Resource {
     }
 
     /**
+     * PUT bike with ID
+     */
+    @PUT
+    @Path('{id: \\d+}')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response putByID (@PathParam('id') Integer id, @Valid Bike userBike) {
+        Response returnResponse
+        Integer idCheck = bikeDAO.idCheck(id)
+        Bike currentBike
+
+        if (!idCheck) {
```

Comment:
Instead of first checking the id and then a few lines later calling bikeDAO.getById(), why not just first use: Bike currentBike = bikeDAO.getById() ? Your check in line 108, could just check whether or not your currentBike is null.